src/components/entry/EntryDetails.js

```diff
@@ -78,29 +95,41 @@ const Description = styled.section`
 
 const EntryDetails = () => {
   const { id } = useParams()
-
+  const { setView } = useContext(MapContext)
   const [locationData, setLocationData] = useState()
+  const [address, setAddress] = useState('')
   const [typesData, setTypesData] = useState()
+  const history = useHistory()
+
+  const { setSelectedLocation } = useContext(SearchContext)
 
   useEffect(() => {
     async function fetchEntryDetails() {
       // Show loading between entry selections
-      setLocationData(null)
-
       const locationData = await getLocationById(id)
+      const streetAddress = await getStreetAddress(
+        locationData.lat,
+        locationData.lng,
+      )
+
       const typesData = await Promise.all(
         locationData.type_ids.map(getTypeById),
       )
-
+      setAddress(streetAddress)
       setLocationData(locationData)
       setTypesData(typesData)
     }
     fetchEntryDetails()
   }, [id])
 
   const _handleAddressClick = () => {
-    // TODO: handle address click
-    console.log('Map Button Clicked')
+    history.push('/map')
+    const tempViewState = {
+      center: { lat: locationData.lat, lng: locationData.lng },
+      zoom: 18,
+    }
```

Comment:
Can you change the getGeolocationBounds behavior to this as well, then use one function?